tests/test_repository_tool.py

```diff
@@ -550,6 +557,168 @@ def test_get_filepaths_in_directory(self):
 
 
 
+  def test_writeall_abstract_storage(self):
+    # Test creation of a TUF repository with a custom storage backend to ensure
+    # that function relying on a storage backend being supplied operate
+    # correctly
+
+
+    class TestStorageBackend(securesystemslib.storage.StorageBackendInterface):
+      """
+      An implementation of securesystemslib.storage.StorageBackendInterface
+      which mutates filenames on put()/get(), such that trying to read the
```

Comment:
Very nice way of testing this. Would you mind adding one sentence that elaborates on the mutation, i.e. `filename` in memory vs. `filename + ".tst"` on disk?

Reply:
Updated to the following in a force-pushed change
> An implementation of securesystemslib.storage.StorageBackendInterface which mutates filenames on put()/get(), translating filename in memory to filename + '.tst' on-disk, such that trying to read the expected/canonical file paths from local storage doesn't find the TUF metadata files.